### PR TITLE
3 - Baseline styles using Tailwind utility classes

### PR DESCRIPTION
**Pull request by airtwo**
This removes all custom CSS declarations on the /web side of things, and adds utility classes from the Tailwind library to some of the default components. The goal is to add only enough so that the default Yote UI is digestible, but not too opinionated, right out of the box.

### CODE REVIEW

File: web/src/global/components/navigation/DefaultNav.jsx [hunk 1]
```diff
@@ -35,26 +35,25 @@ const DefaultNav = () => {
...
+        {!loggedInUser ?
+          <ul class="list-none p-0 flex flex-col md:flex-row md:items-center">
+            <li><NavLink to={{ pathname: "/user/login", state: { from: location } }} className="p-2 block">Sign in</NavLink></li>
+            <li><NavLink to={{ pathname: "/user/register", state: { from: location } }} className="p-2 block">Register</NavLink></li>
+          </ul>
+          :
+          <ul class="list-none p-0 flex flex-col md:flex-row md:items-center">
+            <li><NavLink to="/user/profile">My profile</NavLink></li>
+            <button onClick={handleLogout}>Logout</button>
+          </ul>
```
Comment (WesJourdan) [inline web/src/global/components/navigation/DefaultNav.jsx]: `class` should be `className` on the `ul` elements.

File: web/src/global/components/helpers/WaitOn.jsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const WaitOn = ({
     // still waiting for data
     if(isLoading) return fallback
     // fetch returned empty
-    if(isEmpty) return <div>No data found</div>
+    if(isEmpty) return <div class="p-8"><p className="text-gray-500 italic text-center">No data found</p></div>
```
Comment (WesJourdan) [inline web/src/global/components/helpers/WaitOn.jsx]: `class` -> `className`

File: web/src/global/components/landing/Landing.jsx [hunk 3]
```diff
@@ -4,8 +4,10 @@ import DefaultLayout from '../layouts/DefaultLayout';
 const Landing = () => {
   return (
     <DefaultLayout title="Yote React Web">
-      <h1> Welcome to Yote!</h1>
-      <h4> Check out the docs on <a href="https://fugitivelabs.github.io/yote/"> GitHub </a></h4>
+      <section class="p-2">
```
Comment (WesJourdan) [inline web/src/global/components/landing/Landing.jsx]: `class` -> `className`

File: web/src/global/components/layouts/DefaultLayout.jsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ const DefaultLayout = ({ ...props }) => {
     <div>
       {/* <Helmet title={props.title || "Yote App"}/> */}
       <DefaultNav />
-      <main>
+      <main class="py-20">
```
Comment (WesJourdan) [inline web/src/global/components/layouts/DefaultLayout.jsx]: `class` -> `className`

File: web/src/resources/product/views/ProductList.jsx [hunk 5]
```diff
@@ -19,19 +19,23 @@ const ProductList = () => {
...
-        pagination={pagination}
-        className={`${productQuery.isFetching ? 'opacity-50' : ''}`}
-      >
-        <WaitOn query={productQuery} fallback={<Skeleton count={pagination.per} />}>
-          {products?.map(product => <ProductListItem key={product._id} id={product._id} />)}
-          {/* {ids?.map(productId => <ProductListItem key={productId} id={productId} />)} */}
-        </WaitOn>
-      </PaginatedList>
+      <section class="max-w-screen-lg card">
+        <header class="flex items-center justify-between border-solid border-t-0 border-l-0 border-r-0 border-b p-2">
```
Comment (WesJourdan) [inline web/src/resources/product/views/ProductList.jsx]: `class` -> `className`

File: web/src/resources/user/components/UserRegisterForm.jsx [hunk 6]
```diff
@@ -23,9 +23,9 @@ const UserRegisterForm = ({
   }
 
   return (
-    <div>
+    <div class="card max-w-lg p-4 mt-16">
```
Comment (WesJourdan) [inline web/src/resources/user/components/UserRegisterForm.jsx]: `class` -> `className`

**Review by WesJourdan — COMMENTED** [review]
Just one comment a bunch of times.

File: web/src/global/components/navigation/DefaultNav.jsx [hunk 7]
```diff
@@ -35,26 +35,25 @@ const DefaultNav = () => {
...
-          <li><NavLink to={{ pathname: "/user/login", state: { from: location } }}>Sign in</NavLink></li>
-          <li><NavLink to={{ pathname: "/user/register", state: { from: location } }}>Register</NavLink></li>
-        </ul>
-        :
-        <ul>
-          <li><NavLink to="/user/profile">My profile</NavLink></li>
-          <button onClick={handleLogout}>Logout</button>
+    <header className="border-b border-solid shadow-sm bg-white">
+      <div className="p-2 flex flex-col md:flex-row md:justify-between md:items-center">
+        <ul class="list-none p-0 flex flex-col md:flex-row md:items-center">
```
Comment (ffugly) [inline web/src/global/components/navigation/DefaultNav.jsx]: `class` ->  `className`

File: web/src/global/components/navigation/DefaultNav.jsx [hunk 8]
```diff
@@ -35,26 +35,25 @@ const DefaultNav = () => {
...
+          <li><NavLink to="/" className="p-2 block">Home</NavLink></li>
+          <li><NavLink to="/products" className="p-2 block">Products</NavLink></li>
         </ul>
-      }
-      <hr/>
+        {!loggedInUser ?
+          <ul class="list-none p-0 flex flex-col md:flex-row md:items-center">
```
Comment (ffugly) [inline web/src/global/components/navigation/DefaultNav.jsx]: `class` -> `className`

File: web/src/global/components/navigation/DefaultNav.jsx [hunk 9]
```diff
@@ -35,26 +35,25 @@ const DefaultNav = () => {
...
+            <li><NavLink to={{ pathname: "/user/login", state: { from: location } }} className="p-2 block">Sign in</NavLink></li>
+            <li><NavLink to={{ pathname: "/user/register", state: { from: location } }} className="p-2 block">Register</NavLink></li>
+          </ul>
+          :
+          <ul class="list-none p-0 flex flex-col md:flex-row md:items-center">
```
Comment (ffugly) [inline web/src/global/components/navigation/DefaultNav.jsx]: `class` -> `className`

File: web/src/index.css [hunk 10]
```diff
@@ -1,3 +1,35 @@
...
+  a { @apply text-gray-800; }
+  
+  a:hover { @apply bg-green-400; }
+  
+  label { @apply block; }
+  
+  button { @apply m-0; }
+  
+  ul { @apply m-0 p-0; }
+}
```
Comment (ffugly) [inline web/src/index.css]: Can you add a comment here stating what we're trying to accomplish by adding these?

File: web/src/index.css [hunk 11]
```diff
@@ -1,3 +1,35 @@
...
+
+
+@layer components {
+  .btn {
+    @apply text-sm p-2 px-8 rounded-full border border-solid bg-white text-gray-800 border border-gray-800 cursor-pointer hover:bg-green-400 no-underline font-semibold;
+  }
+  
+  .card {
+    @apply border border-solid bg-white shadow-sm rounded-sm mx-auto;
+  }
```
Comment (ffugly) [inline web/src/index.css]: add comment

File: web/tailwind.config.js [hunk 12]
```diff
@@ -5,7 +5,34 @@ module.exports = {
...
+          200: '#90FFC5',
+          300: '#62FFAD',
+          400: '#11FF83',
+          500: '#0DF07A',
+          600: '#0AD46B',
+          700: '#07A452',
+          800: '#057339',
+          900: '#005127',
+        }
+      }
```
Comment (ffugly) [inline web/tailwind.config.js]: Can you add a brief comment here stating what this does?

**Review by ffugly — CHANGES_REQUESTED** [review]

**Comment by ffugly** [comment]
@airtwo can you please add a description to this PR?

**Comment by grantfowler** [comment]
also, this is purely just our internal style, but can you update the PR title to be "3 - baseline styles"? in a more generalized version, "$base branch - $changes"

File: web/src/index.css (hunk 11 above)
Comment (airtwo) [reply web/src/index.css]: I pulled these back out into just utilities in each component, but left the button class there as an example to go along with the comment.

File: web/tailwind.config.js (hunk 12 above)
Comment (airtwo) [reply web/tailwind.config.js]: Going to pull these specific colors out, as they were more for illustrating an example in my post. Outside of example, for which Tailwind docs should be the source anyways, I don't see a case for including these in a generic starting point.

**Review by ffugly — APPROVED** [review]

**Review by WesJourdan — APPROVED** [review]

**Pull request merged**